src/NLog/Targets/WebServiceTarget.cs

```diff
@@ -412,9 +429,18 @@ protected override string GetFormattedContent(string parametersContent)
 
             protected override string GetFormattedParameter(MethodCallParameter parameter, object value)
             {
+                string parameterValue = Convert.ToString(value, CultureInfo.InvariantCulture);
+                string valueString = string.Empty;
+                if (!string.IsNullOrEmpty(parameterValue))
+                {
+                    var sb = new StringBuilder(Math.Max(parameterValue.Length + 20, Target.Parameters.Count > 1 ? 256 : 0));
+                    UrlHelper.EscapeDataEncode(parameterValue, sb, encodingFlags);
+                    valueString = sb.ToString();
+                }
+
                 return string.Format("{0}={1}",
```

Comment:
Could it not be changed it into (Could probably be optimized even more if re-using the StringBuilder):

if (string.IsNullOrEmpty(parameterValue))
{
    return string.Concat(parameter.Name, "=");
}
else
{
    var sb = new StringBuilder(parameter.Name.Length + parameterValue.Length + 20);
    sb.Append(parameter.Name);
    sb.Append("=");
    UrlHelper.EscapeDataEncode(parameterValue, sb, encodingFlags);
    return sb.ToString();
}